Replace v0.42 with v1.0

turbolent opened the pull request:
Now that Flow Testnet has been updated to Cadence 1.0, and the update of Mainnet is less than a week out, let's remove the current/soon-to-be-outdated v0.42 content (the content at the root).

"Remove" it by replacing it with the v1.0 content. 

Please give it a try in the preview: https://cadence-lang-org-git-bastian-remove-v042-onflow.vercel.app/

SupunS reviewed (approved):
So this simply removes `docs` directory and renamed the `versioned_docs/version-1.0` to `docs`?

turbolent commented:
@SupunS effectively, exactly! 

I didn't actually do a move, but did a bit of a "safer" "copy": I diffed `docs` and `versioned_docs` (`diff -ruq docs versioned_docs/version-1.0`), and removed the old files that exist in the former but not the latter (`Only in docs: ...`, then copied `versioned_docs` to `docs`, and finally removed `versioned_docs`.

Finally I removed the `1.0` version from the config
